### PR TITLE
trying signup using basket and exact-target

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: mixins/form.jsx [hunk 1]
```diff
@@ -137,7 +137,10 @@ module.exports = {
   },
   submit: function(action, props, callback) {
     props.locale = this.props.locales[0];
-    props.currency = this.state.currency.code;
+    var currency = this.state.currency;
+    if (currency) {
+      props.currency = currency.code;
+    }
```
Comment (ScottDowne) [inline mixins/form.jsx]: This was just a straight up bug that was blocking signups.

state.currency only exists for donations, singup was throwing a console error.

File: mixins/form.jsx [hunk 2]
```diff
@@ -350,15 +353,15 @@ module.exports = {
     }
   },
   signup: function(validate, props) {
-    this.setState({
-      submitting: true
-    });
```
Comment (ScottDowne) [inline mixins/form.jsx]: This was creating a waiting spinner even if we know it was going to fail. I moved it down to ensure we only start the spinner if basic validation passes.

**Comment by ScottDowne** [comment]
R?

**Comment by ScottDowne** [comment]
I tried this with ten minute email, and it worked.

So likely it was the fact I was already signed up that was causing this to succeed but do nothing.

File: routes/index.js [hunk 3]
```diff
@@ -7,22 +7,24 @@ var routes = {
...
-      url = url + '-' + transaction.locale;
-    }
-    httpRequest({
+    httpRequest.post({
       url: url,
-      method: 'POST',
       form: {
-        'opt_in': '1',
+        format: 'html',
+        lang: transaction.locale,
```
Comment (adamlofting) [inline routes/index.js]: This is fine, so long as we're only showing the sign-up thank you page for `en` and `de` donors. We haven't setup mailing list for other locales.

**Comment by adamlofting** [comment]
R+

File: routes/index.js (hunk 3 above)
Comment (ScottDowne) [inline routes/index.js]: Yup, we handle that here: https://github.com/mozilla/donate.mozilla.org/blob/master/pages/thank-you.jsx#L35

**Pull request merged**